examples/lib/stories/collision_detection/multiple_shapes.dart

```diff
@@ -0,0 +1,242 @@
+      } else {
+        hitPaint..color = hitColor;
+      }
+    };
+  }
+
+  @override
+  void render(Canvas canvas, Paint paint) {
+    super.render(canvas, hitPaint);
+  }
+}
+
+class CollidableSnowman extends MyCollidable {
+  CollidableSnowman(Vector2 position, Vector2 size, Vector2 velocity)
+      : super(position, size, velocity) {
+    rotationSpeed = 0.2;
+    final top = SnowmanPart(0.4, Vector2(0, -0.8), Colors.red);
+    final middle = SnowmanPart(0.6, Vector2(0, -0.3), Colors.yellow);
+    final bottom = SnowmanPart(1.0, Vector2(0, 0.5), Colors.green);
+    addShape(top);
+    addShape(middle);
+    addShape(bottom);
+  }
+}
+
+class MultipleShapes extends BaseGame
+    with HasCollidables, HasDraggableComponents {
+  final TextConfig fpsTextConfig = TextConfig(
+    color: BasicPalette.white.color,
+  );
+
+  @override
+  Future<void> onLoad() async {
+    final screen = ScreenCollidable();
+    final snowman = CollidableSnowman(
+      Vector2.all(150),
+      Vector2(100, 200),
+      Vector2(-100, 100),
+    );
+    MyCollidable lastToAdd = snowman;
+    add(screen);
+    add(snowman);
+    var totalAdded = 1;
+    while (totalAdded < 20) {
+      lastToAdd = createRandomCollidable(lastToAdd);
+      final lastBottomRight =
+          lastToAdd.toAbsoluteRect().bottomRight.toVector2();
+      if (screen.containsPoint(lastBottomRight)) {
+        add(lastToAdd);
+        totalAdded++;
+      } else {
+        break;
+      }
+    }
+  }
+
+  final _rng = Random();
+  final _distance = Vector2(100, 0);
+
+  MyCollidable createRandomCollidable(MyCollidable lastCollidable) {
+    MyCollidable collidable;
+    final collidableSize = Vector2.all(50) + Vector2.random(_rng) * 100;
+    final isXOverflow = lastCollidable.position.x +
+            lastCollidable.size.x / 2 +
+            _distance.x +
+            collidableSize.x >
+        size.x;
+    var position = _distance + Vector2(0, lastCollidable.position.y + 200);
+    if (!isXOverflow) {
+      position = (lastCollidable.position + _distance)
+        ..x += collidableSize.x / 2;
+    }
+    final velocity = Vector2.random(_rng) * 200;
+    final rotationSpeed = 0.5 - _rng.nextDouble();
+    final shapeType = Shapes.values[_rng.nextInt(Shapes.values.length)];
+    switch (shapeType) {
+      case Shapes.circle:
+        collidable = CollidableCircle(position, collidableSize, velocity);
+        break;
+      case Shapes.rectangle:
+        collidable = CollidableRectangle(position, collidableSize, velocity)
+          ..rotationSpeed = rotationSpeed;
+        break;
+      case Shapes.polygon:
+        collidable = CollidablePolygon(position, collidableSize, velocity)
+          ..rotationSpeed = rotationSpeed;
+        break;
+    }
+    return collidable;
+  }
+
+  @override
+  void render(Canvas canvas) {
+    super.render(canvas);
+    fpsTextConfig.render(
+      canvas,
+      '${fps(120).toStringAsFixed(2)}fps',
+      Vector2(0, size.y - 24),
+    );
```

Comment:
@erickzanardo I think you mentioned adding a component for this? I think it's a good idea